projects/components/src/action-menu/action-menu.component.ts

```diff
@@ -245,6 +245,17 @@ export class ActionMenuComponent<R, T> {
             });
     }
 
+    /**
+     * The visibility of actions is dependent on their availability call back responses and in VCD application, some of the actions
+     * availability call back response is dependent on closure variables. However, we don't call those call backs every time those closure
+     * variables are updated, for example by asynchronous requests. This has a side effect of actions visibility not getting updated when
+     * those closure variables are updated. So, this convenience method is to make it clear for the user that such side effect exists and
+     * this method will re-trigger the availability call backs of actions
+     */
+    updateDisplayedActions(): void {
+        this.actions = [...this.actions];
```

Comment:
This feels like kind-of a hacky way to trigger an update to the calculation of the actions. What if you extract `set actions` to a method called something like `private refreshActions(actions: ActionItem<R, T>[])` and then set actions calls this method, and updateDisplayedActions also calls this method?